tutorials/s4sdk-caching/s4sdk-caching.md

```diff
@@ -51,268 +56,136 @@ If each of these options apply to your use case, then we highly recommend the ca
-                                BusinessPartner.FIRST_NAME,
-                                BusinessPartner.IS_MALE,
-                                BusinessPartner.IS_FEMALE,
-                                BusinessPartner.CREATION_DATE)
-                        .filter(BusinessPartner.BUSINESS_PARTNER_CATEGORY.eq(category))
-                        .orderBy(BusinessPartner.LAST_NAME, Order.ASC)
-                        .execute();
-
-        return businessPartners;
-    }
-
-    @Override
-    protected List<BusinessPartner> getFallback() {
-        logger.warn("Fallback called because of exception:", getExecutionException());
-        return Collections.emptyList();
-    }
+    // Assure an empty list is returned as fallback
+    when()
+            .get("/businesspartners")
+            .then()
+            .statusCode(200)
+            .contentType(ContentType.JSON)
+            .body("", Matchers.hasSize(0));
 }
-
 ```
-You see the following changes, compared to the simple cached-command example:
-
-- `The constructor features a parameter, a `String` representation of a category parameter. This parameter is used in the OData query.`
-- The cache instance now defines:
-- `maximum item size of 100`
-- Up to a hundred business partner queries and their responses will be cached
-- `expiration after `5 minutes`
-- The item associated to any key which is older than five minutes will be re-requested by the command`
-- `concurrency level of `10`
-- This value provides a hint for the underlying caching API to estimate the number of threads trying to write into the cache at the same time. Concurrent reading access will be unaffected from this setting`
-
-
-
-- `These methods have been overridden:`
-- `getCommandCacheKey()`
-- It does append the provided category to the `cacheKey`, thus making it distinguishable from the same service calls but with different category parameter
-- `getFallback()`
-- As before, in case the `runCachable` procedure fails in any way, we provide a fallback solution. In this example we return an empty list object
 
-To use the new class in the servlet, just read a parameter from the "`request`" input variable and use it for instantiating this `parameterized` command class. Or you could create different servlet endpoints for persons (category "1"), organizations (2) and groups (3), which supply the corresponding category to the newly created `parameterized` command. Feel free to try it out!
+Executed on it's own, this test will still pass as it did before. However, if `testService` is run prior to this test the result will be cached and `testWithFallback` will fail since the cached data is returned instead of falling back to an empty list. By changing the test we avoided such issues since our new test does not interfere with other tests. While being sufficient for this tutorial, in a productive setting one should implement a more sophisticated and robust test setup, where caches are invalidated between tests.
```

Comment:
One could also explicitly invalidate the cache prior to executing the fallback test (with drawbacks for parallel execution of tests)

Reply:
discussed offline